tests/unittests/typeAToTypeBFunctionConverterTest.cpp

```diff
@@ -439,6 +439,134 @@ TEST_P(AllBackends, int64IConversionFloatToFloat16) {
+  Module mod;
+  Function *F = mod.createFunction("test");
+  Context ctx;
+
+  auto *input =
+      mod.createPlaceholder(ElemKind::FloatTy, {20, 13}, "Input", false);
+  auto *output =
+      mod.createPlaceholder(ElemKind::FloatTy, {20, 10}, "Output", false);
+
+  auto *weights = mod.createConstant(
+      mod.uniqueType(ElemKind::FloatTy, {13, 10}), "weights");
+  auto *bias =
+      mod.createConstant(mod.uniqueType(ElemKind::FloatTy, {10, 20}), "bias");
+
+  // This save is just to test that we do the right thing for constants with
+  // more than one use.
+  auto *saveBias = F->createSave("saveBias", bias);
+  TypeRef FCTy = mod.uniqueType(ElemKind::FloatTy, {20, 10});
+  auto *FC = F->createFullyConnected("FC", input, weights, bias, FCTy);
+  auto *ReLU = F->createRELU("ReLU", FC, FC->getType(0));
+  auto *result = F->createSave("save", ReLU, output);
+
+  size_t origGraphSize = F->getNodes().size();
+
+  TypeAToTypeBFunctionConverter converter(*F, ElemKind::FloatTy,
+                                          ElemKind::Float16Ty);
+  converter.convert();
+  bool changed = converter.optimizeConversions();
+  EXPECT_TRUE(changed);
+
+  // We should have 2 more nodes:
+  // 1 conversion float to float16 for the input of FC
+  // 1 conversion float16 to float for the result of ReLU.
+  EXPECT_EQ(F->getNodes().size(), origGraphSize + 2);
+  // Make sure the save node is still in the function and is unchanged.
+  EXPECT_TRUE(std::find(F->getNodes().begin(), F->getNodes().end(), *result) !=
+              F->getNodes().end());
+  EXPECT_EQ(result->getOutput(), NodeValue(output, 0));
+  // Check that the save is fed from a conversion from float16 to float.
+  auto *convertedBackReLURes =
+      llvm::dyn_cast<ConvertToNode>(result->getInput());
+  ASSERT_NE(convertedBackReLURes, nullptr);
+  EXPECT_EQ(convertedBackReLURes->getElementType(0), ElemKind::FloatTy);
+  auto *convertedReLU =
+      llvm::dyn_cast<ReluNode>(convertedBackReLURes->getInput());
+  ASSERT_NE(convertedReLU, nullptr);
+  EXPECT_EQ(convertedReLU->getElementType(0), ElemKind::Float16Ty);
+
+  // Check that the ReLU is fed directly by FC float16.
+  auto *convertedFC =
+      llvm::dyn_cast<FullyConnectedNode>(convertedReLU->getInput());
+  ASSERT_NE(convertedFC, nullptr);
+  EXPECT_EQ(convertedFC->getElementType(0), ElemKind::Float16Ty);
+  // Check that the input of FC is a convertTo node from "input" from float to
+  // Float16Ty.
+  auto *convertedFCInput =
+      llvm::dyn_cast<ConvertToNode>(convertedFC->getInput());
+  ASSERT_NE(convertedFCInput, nullptr);
+  EXPECT_EQ(convertedFCInput->getElementType(0), ElemKind::Float16Ty);
+  EXPECT_TRUE(llvm::isa<Placeholder>(convertedFCInput->getInput()));
+  EXPECT_EQ(convertedFCInput->getInput().getElementType(), ElemKind::FloatTy);
+  EXPECT_EQ(convertedFCInput->getInput().getNode(), input);
+
+  // Check that the weights have been updated to float16.
+  auto *convertedFCWeights =
```

Comment:
As noted before, perhaps you can set the value of weights earlier in this test and then check that the value was actually converted?

Reply:
Done